src/editors/ShelfEditor/index.js

```diff
@@ -70,33 +69,62 @@ class ShelfEditor extends React.Component {
+        <form className="ShelfEditor__inner" onSubmit={this.handleSave}>
           <div className="ShelfEditor__text-field">
             <label htmlFor="shelf-title">Título da Prateleira</label>
-            <input id="shelf-title" className="form-control" name="title" type="text"
-                   value={this.state.title} onChange={this.changeValue.bind(this)}
-                   placeholder="Ex: Destaques, Promoção"/>
+            <input
+              id="shelf-title"
+              className="form-control"
+              name="title"
+              type="text"
+              value={this.state.title}
+              onChange={this.changeValue}
+              placeholder="Ex: Destaques, Promoção"
+            />
           </div>
           <div className="ShelfEditor__text-field">
             <label htmlFor="shelf-category">Categoria</label>
-            <input id="shelf-category" className="form-control" name="category" type="text"
-                   value={this.state.category} onChange={this.changeValue.bind(this)}
-                   placeholder="Ex: camiseta, cadeira"/>
+            <input
+              id="shelf-category"
+              className="form-control"
+              name="category"
+              type="text"
+              value={this.state.category}
+              onChange={this.changeValue}
+              placeholder="Ex: camiseta, cadeira"
+            />
           </div>
           <div className="ShelfEditor__quant-field">
             <label htmlFor="shelf-quantity">Quantidade de Produtos na Prateleira</label>
             <div className="ShelfEditor__quant-selector">
-              <button className="ShelfEditor__quant-button--left" onTouchTap={this.decrementQuantity.bind(this)}> - </button>
-                <input id="shelf-quantity" className="ShelfEditor__quant-selector__input" name="quantity" type="text"
-                       value={this.state.quantity} onChange={this.changeQuantity.bind(this)}/>
-              <button className="ShelfEditor__quant-button--right" onTouchTap={this.incrementQuantity.bind(this)}> + </button>
+              <button
+                className="ShelfEditor__quant-button--left"
+                onClick={this.decrementQuantity}
+              >
+                -
+              </button>
+              <input
+                id="shelf-quantity"
+                className="ShelfEditor__quant-selector__input"
+                name="quantity"
+                type="text"
+                value={this.state.quantity}
+                onChange={this.changeQuantity}
+              />
+              <button
+                className="ShelfEditor__quant-button--right"
+                onClick={this.incrementQuantity}
+              >
+                +
+              </button>
             </div>
           </div>
         </form>
 
-        <ActionBar onSave={this.handleSave.bind(this)}/>
+        <ActionBar onSave={this.handleSave} />
```

Comment:
The ActionBar component expects an **"id"** prop:

```js
<ActionBar id={this.props.componentProps.id} onSave={this.handleSave.bind(this)}/>
```

https://github.com/vtex-apps/banner/blob/master/src/editors/BannerEditor/index.js#L61